gh-64978: Add `chown()` to `pathlib.Path` (#31212)
python/cpython: Doc/library/pathlib.rst

   .. versionchanged:: 3.10
      The *follow_symlinks* parameter was added.

.. method:: Path.chown(uid, gid, *, follow_symlinks=True)

   Change the file ownership, like :func:`os.chown`.

   This method normally follows symlinks. Some Unix flavours support changing
   permissions on the symlink itself; on these platforms you may add the

>    argument ``follow_symlinks=False``, or use :meth:`~Path.lchown`.

Reviewer: This mentions `Path.lchown()`, but it doesn't exist (in the code or the PR).

>    argument ``follow_symlinks=False``, or use :meth:`~Path.lchown`.

Reviewer 2: Why do we need two ways to do the same thing?

Author: I do not think, that we essentially need both functions. But since both are available in `os` and both `chmod` and `lchmod` are available in `pathlib` I adapted this behavior for `chown`.

   ::

      >>> p = Path('setup.py')
      >>> p.stat().st_uid
      1000
      >>> p.stat().st_gid
      1000
      >>> p.chown(1, 20)
      >>> p.stat().st_uid
      1
      >>> p.stat().st_gid
      20

>    .. availability:: Unix.
> 
> 

Reviewer 3: ```suggestion
   .. availability:: Unix.

   .. versionadded:: 3.12
 
 
```

Reviewer 2: Too late now, it'll have to be 3.12.

Reviewer 3: Edited 3.11 --> 3.12.

   .. versionchanged:: 3.10
      The *follow_symlinks* parameter was added.

.. method:: Path.chown(uid, gid, *, follow_symlinks=True)

   Change the file ownership, like :func:`os.chown`.

>    This method normally follows symlinks. Some Unix flavours support changing

Reviewer: ```suggestion
   This method follows symlinks by default. Some Unix flavours support changing
```

   symbolic link's mode is changed rather than its target's.

.. method:: Path.lchown(uid, gid, *, dir_fd=None)

>    Like :meth:`Path.chown` but, if the path points to a symbolic link, the

Reviewer 2: ```suggestion
   Like :meth:`Path.chown`, but if the path points to a symbolic link, the
```

:func:`os.path.abspath`                :meth:`Path.absolute` [#]_
:func:`os.path.realpath`               :meth:`Path.resolve`
:func:`os.chmod`                       :meth:`Path.chmod`

> :func:`os.chowm`                       :meth:`Path.chown`

Reviewer 4: Spelling error on `os.chown`
